### DISCUSSION TITLE
More consideration for TPR

### DISCUSSION THREAD

**Discussion by HeavenCrafter** [EssentialsX/Essentials#4174] [Ideas and Feature Suggestions]
I've been using EssentialsX built in TPR as my full replacement for my server's random teleport feature considering that I don't really need any sort of advance features. It's near perfect. However, I have come across a sort of a feature that is lacking. My server is using a premade city made with really tall buildings and especially for newer players who simply just want to get down quick, most of the time, they will get themselves killed expecting that there is some sort of "no fall damage grace period". It has become sort of an annoyance that players aren't able to get down tall buildings without falling to their deaths everytime they randomly teleport.

With that said, it'd be really great if there was some sort of configuration/workaround for players using TPR. Say like a config option to enable a 10 seconds of no fall damage to prevent new players who just falls to their death.

**Comment by triagonal** [comment]
Personally, I'm not sure this would be within scope for EssentialsX - especially considering that EssentialsX already ensures that the user is in a safe location - though the dev team might have different opinions. There are a couple options that already exist though:

1. EssentialsX has a built in `teleport-invulnerability` config option that you could make use of, though you can't pick and choose which teleports trigger the effect.

2. EssentialsX provides a [UserRandomTeleportEvent](https://jd-v2.essentialsx.net/net/ess3/api/events/userrandomteleportevent) which plugin (and Skript) developers can use to detect when a random teleport takes place and then to take an action as necessary.

**Reply by HeavenCrafter** [reply]
`teleport-invulnerability` might be the best option in my case considering that as long as players get to live for whenever they are randomly teleported on a tall building and just wanna get down by simply jumping off, it's definitely an option. Though maybe for other players, who may deal with a similar scenario as I am right now, might prefer to have a bit of an extra flexibility for it. But for my case, this option works the best for me. Thanks for the quick response and thanks for letting me know about this category of options in the config.

**Comment by molor** [comment]
Why we can't just add the `max-location-y` or something similar to random teleport settings file to select random coordinates lower that specified?
